scipy/signal/tests/test_filter_design.py

```diff
@@ -2716,14 +2699,15 @@ def test_bandpass(self):
             6.521767004237027e-01 - 6.744414640183752e-01j,
             ]
         k2 = 3.398854055800844e-08
-        assert_array_equal(z, z2)
+        z2 = np.asarray(z2, dtype=z.dtype)
+        xp_assert_equal(z, z2)
```

Comment:
I would probably prefer to just `check_dtype=False` if we can't derive the dtype from the inputs (or can we?)